### PR TITLE
fix(skills): parse YAML block scalars in SKILL.md frontmatter

### LINKED ISSUE #1907: SKILL.md YAML block scalar description not parsed correctly
## Description

When a SKILL.md frontmatter `description` field uses YAML block scalar syntax (`>` folded or `|` literal), the parser stores the indicator character (e.g. `>`) as the description value and ignores the indented content entirely.

For example:

```yaml
description: >
  This is a multi-line
  skill description
```

The parsed description becomes `">"` instead of `"This is a multi-line skill description"`.

## Steps to reproduce

1. Create a SKILL.md with a block-scalar description:
   ```markdown
   ---
   name: test-skill
   description: >
     line one
     line two
   ---
   This is the skill body
   ```

2. Run `/skills` in DeepSeek TUI to list loaded skills, or load the skill via `load_skill`
3. Observe the description displayed for this skill — it shows `>` instead of the actual text

Alternatively, call `SkillRegistry::parse_skill` directly in a unit test to reproduce.

## Expected behavior

- `>` (folded): content lines joined with spaces → `"line one line two"`
- `|` (literal): content lines joined with newlines → `"line one\nline two"`
- `>-` / `|-` (strip chomp): trailing blank lines removed
- `>+` / `|+` (keep chomp): trailing blank lines preserved
- Single-line descriptions (non-block-scalar): unchanged backward compat

## Actual behavior

Block scalar continuation lines are completely ignored. The `description` field is set to the raw indicator string (e.g. `">"`).

## Impact

Skills authored with multi-line descriptions (using YAML block scalars) show incorrect description text in `/skills`, `load_skill` tool output, and the system prompt's skills context block.

## Fix

A fix has been committed to `feat/skill-yaml-block-scalar` branch (fork at `zlh124/DeepSeek-TUI`). It implements proper YAML block scalar parsing with indentation-based block collection, chomping (strip/clip/keep), folding (space-join) and literal (newline-join) modes, plus 13 unit tests covering all variants.

## Environment

- DeepSeek CLI version: 0.8.40

### PR DESCRIPTION
**Pull request by zlh124**
## Summary

Fixes YAML block scalar parsing in SKILL.md frontmatter descriptions. The parser previously only handled single-line quoted/unquoted values — multi-line descriptions using `>` or `|` were silently parsed as the literal indicator string.

### Changes

**`crates/tui/src/skills/mod.rs`** — extended `SkillRegistry::parse_skill`:

- **Block scalar detection**: recognizes `>`, `>-`, `>+`, `|`, `|-`, `|+` indicators
- **Indentation-based content collection**: continuation lines are gathered by comparing each line's indent against the base indent of the key line (per YAML spec)
- **Three chomping modes**:
  - `strip` (`-`): removes all trailing blank lines
  - `clip` (default): keeps at most one trailing newline
  - `keep` (`+`): preserves all trailing blank lines
- **Folded** (`>`): non-empty lines joined with spaces; empty lines become paragraph breaks (`\n`)
- **Literal** (`|`): lines joined with newlines
- Single-line descriptions remain fully backward compatible

### Tests

13 unit tests added covering all six block-scalar variants plus chomping edge cases:

| Test | Scenario |
|------|----------|
| `parse_skill_folded_block_scalar` | `>` folded block scalar |
| `parse_skill_folded_strip_block_scalar` | `>-` folded + strip via render pipeline |
| `parse_skill_literal_block_scalar` | `\|` literal block scalar |
| `parse_skill_direct_folded_result` | Direct `>` assertion on `Skill` struct |
| `parse_skill_strip_chomp_strips_trailing_empties` | `>-` strips trailing empties |
| `parse_skill_keep_chomp_preserves_trailing_empties` | `>+` preserves trailing empties |
| `parse_skill_clip_chomp_clips_excess_trailing_empties` | `>` clip with 3 trailing empties |
| `parse_skill_clip_chomp_no_trailing_empties` | `>` clip with 0 trailing empties |
| `parse_skill_clip_chomp_one_trailing_empty` | `>` clip with 1 trailing empty |
| `parse_skill_strip_vs_keep_trailing` | `>-` vs `>+` comparison |
| `parse_skill_literal_strip_strips_trailing_newlines` | `\|-` literal strip |
| `parse_skill_literal_keep_preserves_trailing_newlines` | `\|+` literal keep |
| `parse_skill_single_line_description_still_works` | Single-line backward compat |

All pass:

```
test result: ok. 13 passed; 0 failed; 0 ignored; 0 measured
```

### Related issue

Fixes: #1907

### CODE REVIEW

File: crates/tui/src/skills/mod.rs [hunk 1]
```diff
@@ -249,23 +249,110 @@ impl SkillRegistry {
...
+                                continue;
+                            }
+                            let line_indent = raw_line.len() - raw_line.trim_start().len();
+                            if line_indent > base_indent {
+                                block_lines.push(raw_line.trim());
+                                i += 1;
+                            } else {
+                                break;
+                            }
+                        }
```
[the remark this answers is not in the document]
Comment (zlh124) [reply crates/tui/src/skills/mod.rs:302]: fix already

**Comment by zlh124** [comment]
Maybe use a yaml parser is better. but for now, is ok.

**Comment by Hmbown** [comment]
Stewardship note: linked #1907 into v0.8.44 and left it open until this lands. Checks are green right now, and the review focus should be parser scope/backward compatibility rather than UI text only, since the parsed description feeds /skills, load_skill, and model prompt context.

Thanks @zlh124 for the clear repro and tests.

**Comment by zlh124** [comment]
> Stewardship note: linked #1907 into v0.8.44 and left it open until this lands. Checks are green right now, and the review focus should be parser scope/backward compatibility rather than UI text only, since the parsed description feeds /skills, load_skill, and model prompt context.
> 
> Thanks @zlh124 for the clear repro and tests.

谢谢你，鲸鱼兄弟！
